src/js/_enqueues/lib/nav-menu.js

```diff
@@ -288,6 +290,97 @@
 						});
 					});
 					return this;
+				},
+				updateParentDropdown : function() {
+					return this.each(function(){
+						var menuItems = $( '#menu-to-edit li' ),
+							parentDropdowns = $( '.edit-menu-item-parent' );
+
+						$.each( parentDropdowns, function() {
+							var parentDropdown = $(this),
+								$html = '',
+								$selected = '',
+								currentItemID = parentDropdown.closest('li.menu-item').find('.menu-item-data-db-id').val(),
+								currentparentID = parentDropdown.closest('li.menu-item').find('.menu-item-data-parent-id').val(),
+								currentItem = parentDropdown.closest('li.menu-item'),
+								currentMenuItemChild = currentItem.childMenuItems(),
+								excludeMenuItem = [currentItemID];
+
+							if( currentMenuItemChild.length > 0 ){
+								$.each( currentMenuItemChild, function(){
+									var childItem = $(this),
+										childID = childItem.find('.menu-item-data-db-id').val();
+
+									excludeMenuItem.push(childID);
+								});
+							}
+
+							if(currentparentID == 0) $selected = 'selected';
+
+							$html += '<option ' + $selected + ' value="0">No Parent</option>';
+
+							$.each( menuItems, function() {
+								var menuItem = $(this),
+								$selected = '',
+								menuID = menuItem.find('.menu-item-data-db-id').val(),
+								menuTitle = menuItem.find('.edit-menu-item-title').val();
+
+								if(!excludeMenuItem.includes(menuID) ) {
+									if(currentparentID == menuID) $selected = 'selected';
+									$html += '<option ' + $selected + ' value="'+menuID+'">'+menuTitle+'</option>';
+								}
+							});
+
+							parentDropdown.html( $html );
```

Comment:
After updating dropdowns, set focus back to the control that triggered the change & trigger `wp.a11y.speak()` with a simple announcement either "Menu parent updated" or "Menu order updated"